packages/frontend-2/lib/common/composables/toast.ts

```diff
@@ -1,60 +1,95 @@
-import { useTimeoutFn } from '@vueuse/core'
 import type { Optional } from '@speckle/shared'
 import type { ToastNotification } from '@speckle/ui-components'
+import { useTimeoutFn } from '@vueuse/core'
 import { ToastNotificationType } from '@speckle/ui-components'
 import { useSynchronizedCookie } from '~/lib/common/composables/reactiveCookie'
+import { v4 as uuidv4 } from 'uuid'
 
 /**
  * Persisting toast state between reqs and between CSR & SSR loads so that we can trigger
  * toasts anywhere and anytime
  */
 const useGlobalToastState = () =>
-  useSynchronizedCookie<Optional<ToastNotification>>('global-toast-state')
+  useSynchronizedCookie<Optional<ToastNotification[]>>('global-toast-state')
 
 /**
  * Set up a new global toast manager/renderer (don't use this in multiple components that live at the same time)
  */
 export function useGlobalToastManager() {
+  type Timeout = {
+    id: string
+    stop: () => void
+  }
+
   const stateNotification = useGlobalToastState()
 
-  const currentNotification = ref(stateNotification.value)
-  const readOnlyNotification = computed(() => currentNotification.value)
+  const timeouts = ref<Timeout[]>([])
+  const currentNotifications = ref<ToastNotification[]>(
+    Array.isArray(stateNotification.value) ? stateNotification.value : []
+  )
+  const readOnlyNotification = computed(() => currentNotifications.value)
 
-  const dismiss = () => {
-    currentNotification.value = undefined
-    stateNotification.value = undefined
+  // Remove a specific notification from the state
+  const removeNotification = (id: string) => {
+    const index = currentNotifications.value.findIndex((n) => n.id === id)
```

Comment:
As long as there's only ever 1 reference/copy of the toast object, you don't need an ID, you can just compare object references.

```js
const removeNotification = (notif: Notification) => {
  const idx = currentNoficiations.value.findIndex(n => n === notif)
...
}
```

Reply:
Good point, thanks, that should remove the need for all the ID logic :)